src/Commands/play.js

```diff
@@ -0,0 +1,92 @@
+// In JavaScript, classes are closures. Thus, setupGame 
+// can be seen as a class. 
+const setupGame = async function(client, message, args) {
+  if (typeof this.channelList === "undefined") {
+    this.channelList = [];
+  }
+  
+  if (this.channelList.includes(message.channel.id)) {
+    message.channel.send("This channel is taken nya!");
+    return;
+  }
+  this.channelList.push(message.channel.id);
+  
+  let currentValue = 0;
+  let messageId = null;
+  let currentChannel = message.channel.id;
+  message.channel.send(currentValue).then((sent) => {
+    messageId = sent.id;
+    runGameLogic(client, sent);
+  });
+  
+  // Here, this refers to the setupGame class
+  const refreshMessage = (client, message) => {
+    message.edit(currentValue);
+  }
+
+  const updateGameMessage = (client, messageReaction, user) => {
+    if (!user.bot) {
+      if (messageId == messageReaction.message.id) {
+        let hasStopped = false;
+        switch (messageReaction.emoji.toString()) {
+          case "⬅️":
+            currentValue--;
+            break;
+          case "➡️":
+            currentValue++;
+            break;
+          case "❌":
+            messageReaction.message.reactions.removeAll();
+            hasStopped = true;
+            client.emit("stop", messageReaction.message.channel.id);
+            break;
+          default:
+            break;
+        }
+        messageReaction.users.remove(user);
+        if (!hasStopped) {
+          refreshMessage(client, messageReaction.message);
+        }
+      }
+    }
+  }
+
+  const runGameLogic = (client, message) => {
+    message.react("⬅️");
+    message.react("➡️");
+    message.react("❌");
+    /*
+    let activeMessageID = client.variables.gameLogic.currentMessage;
+    message.channel.messages.fetch(activeMessageID)
+      .then(message => {
+      
+    })*/
+  }
+  
+  const f1 = (messageReaction, user) => {
+    updateGameMessage(client, messageReaction, user);
+  }
+  
+  client.on("messageReactionAdd", f1);
```

Comment:
Minor: Is it possible to replace f1 with updateGameMessage on Line 70? Wondering if we can refer to client as a class member instead of having to pass it around as parameter.

(i.e refreshMessage/runGameLogic/updateGameMessage can have one less parameter)

Reply:
Oh right, I can do that. I've edited the code accordingly

Reply:
Bonus question: one would expect that defining this.varname = value would create instance-based variables. However, for some reason, this.channelList behaves like a static variable, whereas let currentValue, let messageId and let currentChannel are instance variables. Is this how people usually do it in JavaScript? I have no idea

Reply:
Update: I figured out what this.varname = value does. If my suspicions are correct, I accidentally added a global-ish property to client called varname